### PR TITLE
z3 v4.10.1 check regarding issue 252

### PR DESCRIPTION
**Pull request by sjunges**
See #252

### CODE REVIEW

File: resources/3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -191,19 +191,25 @@ if(Z3_FOUND)
...
-            set(STORM_HAVE_Z3_OPTIMIZE ON)
-            message (STATUS "Storm - Linking with Z3 (version ${Z3_VERSION}). (Z3 version supports optimization)")
+        if ("${Z3_VERSION}" VERSION_EQUAL "4.10.1")
+            message (WARNING "Storm cannot be linked against Z3 version 4.10.1, see issue 252.")
+            message(WARNING "Storm - Could not link with z3. Building of Prism/JANI models will not be supported.")
+            set(Z3_FOUND FALSE)
         else()
-            message (STATUS "Storm - Linking with Z3 (version ${Z3_VERSION}). (Z3 version does not support optimization)")
+            # Check whether the version of z3 supports optimization
+            if(NOT "${Z3_VERSION}" VERSION_LESS "4.5.0")
```
Comment (volkm) [inline resources/3rdparty/CMakeLists.txt:200]: I know it was already part of the existing code, but is there a reason we did not use `VERSION_GREATER_EQUAL` instead?

**Review by volkm — APPROVED** [review]

File: resources/3rdparty/CMakeLists.txt (hunk 1 above)
Comment (sjunges) [reply resources/3rdparty/CMakeLists.txt:200]: IDK, but it seems too minor to go back and change this.

Comment (volkm) [reply resources/3rdparty/CMakeLists.txt:200]: Sure, just noticed it now and was curious ;). The rest is fine.

Comment (tquatmann) [reply resources/3rdparty/CMakeLists.txt:200]: I think my lack of cmake skills back then are to blame. Sorry :)

**Pull request merged**